src/core/components.py

```diff
@@ -277,6 +277,78 @@ def get_base_component_dict(self):
         return component
 
 
+class InputWithJson(Input):
```

Comment:
Can we inherit the class `InputWithSuggestions` here instead of `Input`? Then we wouldn't need to duplicate a lot of the suggestions related code